### PR TITLE
refactor: model level string literal types to enums

### PR DESCRIPTION
**Pull request by iSCJT**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

Model level string literal types refactored to enums

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Comment by iSCJT** [comment]
I tried doing an interactive rebase on this to tidy it up but was getting into all kinds of conflicts.

File: src/models/user.models.tsx [hunk 1]
```diff
@@ -1,16 +1,13 @@
 /* eslint-disable @typescript-eslint/naming-convention*/
```
Comment (benfurber) [inline src/models/user.models.tsx:1]: Could this be removed now? I think it might be related to the changes you're making?

**Review by benfurber — APPROVED** [review]
Such a great change, nice one @iSCJT. 

One stylistic question and one request please: 
1. Should the enums be imported as types e.g. `import type { IThing } from 'oa-shared'`?
2. Extract child vales from the types. 
e.g. on `src/pages/Howto/Content/Howto/HowtoDescription/HowtoDescription.tsx`:
Do this at the top level:
```const { ACCEPTED } = IModerationStatus```
Then later on you can do:
```howto.moderation === ACCEPTED && (```

File: src/models/user.models.tsx (hunk 1 above)
Comment (iSCJT) [reply src/models/user.models.tsx:1]: Yea I think we can.

**Comment by iSCJT** [comment]
> 2\. const { ACCEPTED } = IModerationStatus

Re: 

1. if you want to use the enum as a value then you can't import it as a type.
2. I've had a look and I don't think this is valid syntax for using enums, just gives you a pointer to the enum.  You could do something like

`const ACCEPTED = IModerationStatus.ACCEPTED`
`const REJECTED = IModerationStatus.REJECTED`
`const AWAITING_MODERATION = IModerationStatus.AWAITING_MODERATION`

etc etc but I don't think it's worth the refactor time.

**Pull request merged**